beacon-chain/blockchain/testing/mock.go

```diff
@@ -357,8 +357,14 @@ func (s *ChainService) IsCanonical(_ context.Context, r [32]byte) (bool, error)
 	return true, nil
 }
 
-// HasInitSyncBlock mocks the same method in the chain service.
-func (s *ChainService) HasInitSyncBlock(rt [32]byte) bool {
+// HasBlock mocks the same method in the chain service.
+func (s *ChainService) HasBlock(ctx context.Context, rt [32]byte) bool {
+	if s.DB == nil {
+		return false
+	}
+	if s.DB.HasBlock(ctx, rt) {
```

Comment:
Does it make sense to switch the order here? the init sync cache should be faster than the DB call.

Reply:
This is a mock function that's only used for tests, so I didn't both to optimize it. I could switch, but I will have to fix more tests because most tests use DB than the initial sync cache, so I will have to make sure those tests have the initial sync cache adequately set up. Let me know if you feel strongly about this. Happy to optimize it

Reply:
Oh no my bad, I thought it was the same behavior in `hasBlockInInitSyncOrDB` which wasn't in the PR, but that function has the right order.